jStorage.github.js

```diff
@@ -310,67 +334,105 @@
+                    if (state === tmpState) {
+                        // Token state are valid, set/change token.
+                        return tmpToken;
+                    }
+                }
+            }
+            return false;
         },
         ensureAuth: function (wrapper, config) {
-            // https://github.com/settings/tokens/new 
+            // we require token today, allow for oauth fetching of token here
+
+            // Because of security reasons, if we have valid token in url, force remove...
+            var token = this.getTokenFromQuery();
+            if (!!token) {
+                // store token for later use
+                window.localStorage.setItem('token', token);
+
+                // Make sure we don't have token in url (as if user copies the url and sends it to friend/or someone else they will be logged in as our user)
+                var search = '&' + window.location.search.substr(1); // replace question mark with '&' char
+                search = search.replace("&token=" + token, '');
+                search = search.replace("&state=" + localStorage.getItem('jStorage.github.tokenState'), '');
+
+                // make sure we remove the temporary tokenState from storage
+                window.localStorage.removeItem('jStorage.github.tokenState');
+                if (search.length === 0) {
+                    // we have removed everything in querystring, reset href without querystring part, we are doing it this way instead of setting 'search' property because setting 'search' property will result in '?' in the end of url.
+                    window.location.href = location.href.replace(location.search, '');
+                } else {
+                    window.location.search = '?' + search.substr(1); // removes first '&' char
+                }
+            }
 
-            // TODO: remove
-            // clientId = 0f1cbd2e4d64c148c51e
-            // proxyAppId = 536981f0-31a7-495d-bb1b-8ef935fdb331
-            //?token=sdfölkasdfdsfasd
+            // If we have a token provided in config we will use that and can there for ignore the rest of below..
+            if (!!this._config.token) {
+                return;
+            }
 
-            this._hasRepo = config && typeof (config.repo) === "string";
-            this._hasToken = config && typeof (config.token) === "string";
-            this._hasTokenService = config && typeof (config.tokenService) === "string";
-            if (this._hasTokenService && !this._hasToken) {
-                // TODO: We should probably make sure that the returned scopes is equal to our requested scope
-                var tmpToken = this.getUrlParameterByName('access_token', window.location);
-                if (!!tmpToken) {
-                    config.token = tmpToken;
-                    this._hasToken = true;
-                }
+            var token = window.localStorage.getItem('token');
```

Comment:
Maybe add some comment here saying: `If we have a token stored in localStorage, we can use that and ignore the rest below`